src/compile/mark/encode/conditional.ts

```diff
@@ -7,21 +7,37 @@ import {parseSelectionPredicate} from '../../selection/parse';
 import {UnitModel} from '../../unit';
 
 /**
- * Return a mixin that includes a Vega production rule for a Vega-Lite conditional channel definition
- * or a simple mixin if channel def has no condition.
+ * Return a VgEncodeEntry that includes a Vega production rule for a scale channel's encoding or guide encoding, which includes:
+ * (1) the conditional rules (if provided as part of channelDef)
+ * (2) invalidValueRef for handling invalid values (if provided as a parameter of this method)
+ * (3) main reference for the encoded data.
  */
-export function wrapCondition<CD extends ChannelDef | GuideEncodingConditionalValueDef>(
-  model: UnitModel,
-  channelDef: CD,
-  vgChannel: string,
-  refFn: (cDef: CD) => VgValueRef
-): VgEncodeEntry {
+export function wrapCondition<CD extends ChannelDef | GuideEncodingConditionalValueDef>({
+  model,
+  channelDef,
+  vgChannel,
+  invalidValueRef,
```

Comment:
FYI, the main change in this file is to add invalidValueRef that caller of this method may pass in.

However, I also switch to object argument so it's easy to make explicit when invalidValueRef is not used.